common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -51,3 +54,105 @@ def perform_destroy(self, instance):
+    permission_classes = (permissions.IsAuthenticated,)
+    queryset = CourseEntitlement.objects.all()
+    serializer_class = CourseEntitlementSerializer
+
+    def _enroll_entitlement(self, entitlement, course_session_key, user):
+        enrollment = CourseEnrollment.enroll(
+            user=user,
+            course_key=course_session_key,
+            mode=entitlement.mode,
+        )
+
+        CourseEntitlement.set_enrollment(entitlement, enrollment)
+
+    def _unenroll_entitlement(self, entitlement, course_session_key, user):
+        CourseEnrollment.unenroll(user, course_session_key, skip_refund=True)
+        CourseEntitlement.set_enrollment(entitlement, None)
+
+    def create(self, request, uuid):
+        course_session_id = request.data.get('course_run_id', None)
+
+        if not course_session_id:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data="The Course Run ID was not provided."
+            )
+
+        # Verify that the user has an Entitlement for the provided Course UUID.
+        try:
+            entitlement = CourseEntitlement.objects.get(uuid=uuid, user=request.user, expired_at=None)
+        except CourseEntitlement.DoesNotExist:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data="The Entitlement for this UUID does not exist or is Expired."
+            )
+
+        # Verify the course run ID is of the same type as the Course entitlement.
+        course_run_valid = False
+        course_runs = get_course_runs_for_course(entitlement.course_uuid)
+        for run in course_runs:
+            if course_session_id == run.get('key', ''):
+                course_run_valid = True
+
+        if not course_run_valid:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data="The Course Run ID is not a match for this Course Entitlement."
+            )
+
+        # Determine if this is a Switch session or a simple enroll and handle both.
+        if entitlement.enrollment_course_run is None:
+            self._enroll_entitlement(
+                entitlement=entitlement,
+                course_session_key=CourseKey.from_string(course_session_id),
+                user=request.user
+            )
+        else:
+            if entitlement.enrollment_course_run.course_id != course_session_id:
+                self._unenroll_entitlement(
+                    entitlement=entitlement,
+                    course_session_key=entitlement.enrollment_course_run.course_id,
+                    user=request.user
+                )
+                self._enroll_entitlement(
+                    entitlement=entitlement,
+                    course_session_key=CourseKey.from_string(course_session_id),
+                    user=request.user
+                )
+
+        return Response(
+            status=status.HTTP_201_CREATED,
```

Comment:
Would it make sense to return a different status for update vs create so that the client knows what happened on the backend? Maybe 200 for the update case and 201 for create? 

Also, if the calls to _enroll/_unenroll fail for some reason, what will be returned?